### ISSUE TITLE
feat(material/checkbox): MatCheckbox is missing support for aria-expanded attribute

### ISSUE THREAD

**Issue by sheogora** [angular/components#28761]
### Feature Description

Based on documentation for [`aria-expanded`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded#associated_roles) one of the roles with which this attribute is associated with is `role=checkbox`. There is currently no way of providing this `aria-expanded` attribute to the input element generated by MatCheckbox component.

Additionally based on the documentation provided `aria-controls` and `aria-owns` are also missing to indicate to the user which section the output of this input will affect. 

Also see further discussion on [extending support to role=radio](https://github.com/w3c/aria/issues/1404)

### Use Case

Checkbox toggle status expands a related section with further options/information. Screen reader should be able to inform the user of any sections expanded/collapsed as a result of their action

Before
![image](https://github.com/angular/components/assets/14091858/bdec082b-82fa-4ca9-87b6-80bf8dcc3c08)

After
![image](https://github.com/angular/components/assets/14091858/3edbfe0b-c0fc-4ca9-b657-658b475da267)
